torchaudio/functional.py

```diff
@@ -566,13 +590,96 @@ def lfilter(waveform, a_coeffs, b_coeffs):
     return torch.min(ones, torch.max(ones * -1, padded_output_waveform[:, (n_order - 1):]))
 
 
+def lfilter_cpp_impl(waveform, a_coeffs, b_coeffs, execution_method):
+    # type: (Tensor, Tensor, Tensor, string) -> Tensor
+    r"""
+    See lfilter documentation. Execution method can be either `element_wise` or `matrix`.
+    """
+
+    # Perform sanity checks, input check, and memory allocation in python
+
+    # Current limitations to be removed in future
+    assert(waveform.dtype == torch.float32 or waveform.dtype == torch.float64)
```

Comment:
Is this implementation in C++ restricting the functionality w/r to the python one?